src/ngx_http_headers_more_headers_out.c

```diff
@@ -752,13 +754,23 @@ ngx_http_headers_more_parse_directive(ngx_conf_t *cf, ngx_command_t *ngx_cmd,
         cmd->headers = NULL;
 
     } else {
+
         h = cmd->headers->elts;
         for (i = 0; i < cmd->headers->nelts; i++) {
 
-            if (ngx_strncasecmp(h[i].key.data, (u_char *) "Set-Cookie",
-                                h[i].key.len) == 0)
-            {
+            handlers = ngx_http_headers_more_set_handlers;
+
+            for (j = 0; handlers[j].name.len; j++) {
+                if (h[i].key.len == handlers[j].name.len
+                    && ngx_strncasecmp(h[i].key.data, handlers[j].name.data,
+                                       h[i].key.len) == 0)
+                {
+                    is_builtin_header = 1;
+                    break;
+                }
+            }
 
+            if (!is_builtin_header) {
```

Comment:
If it is a built-in header and the append flag is true, we should throw an error.

Reply:
Good idea, clearer and easier for users to use